app/models/assignment.rb

```diff
@@ -1266,15 +1283,16 @@ def self.upload_assignment_list(file_format, assignment_data)
     case file_format
     when 'csv'
       result = MarkusCsv.parse(assignment_data) do |row|
-        assignment = self.find_or_create_by(short_identifier: row[0],
-                                            assignment_properties: { repository_folder: row[0] })
+        assignment = self.find_or_create_by(short_identifier: row[0])
         attrs = Hash[DEFAULT_FIELDS.zip(row)]
         attrs.delete_if { |_, v| v.nil? }
         if assignment.new_record?
+          assignment.assignment_properties = AssignmentProperties.new
```

Comment:
Pass in options `token_period: 1, etc.` to the call to `new`, rather than having the three separate statements on lines 1293-1295.